packages/edit-widgets/src/components/widget-areas-block-editor-content/index.js

```diff
@@ -36,11 +54,67 @@ export default function WidgetAreasBlockEditorContent() {
 				>
 					<WritingFlow>
 						<ObserveTyping>
-							<BlockList className="edit-widgets-main-block-list" />
+							{ widgetAreas.map( ( widgetArea ) => {
+								return (
+									<WidgetAreaEditor
```

Comment:
A separate editor per widget area is how it was written initially but for design reasons I believe @jorgefilipecosta refactored this to be a single editor instance:

I agree with you that separate editors are conceptually easier to reason about but a single editor has some advantages too:

 - The global inserter works easily
 - You can move blocks between widget areas.

See #22140 for more details

Reply:
Thank you for this extra context, a single editor instance definitely has upsides - what I found hard was making the blocks live in redux state instead of `const blocks = useState()`. I will keep exploring and possibly follow up with a few questions.

Reply:
I got it to work as a single editor using a similar "ephemeral post" as on the navigation management screen. I think it would also be possible to get rid of the "widget area post" on only have that top level post and widget area entities, but it's a tactical detail.

Reply:
Separate editors also bring other challenges like very hard to have a global undo and redo system, more complex to have block navigation that covers all the widget areas, all the a11y to navigate on widget areas would need to be implemented.